context/context.go

```diff
@@ -60,6 +60,11 @@ func newPooledContext(pool contextPool) Context {
 	return &ctx{pool: pool}
 }
 
+// newContext returns an empty ctx
+func newContext() *ctx {
```

Comment:
nit: Can you change `NewContext` to call `newContext()` instead of `newPooledContext(nil)`?